hugr/src/extension/op_def.rs

```diff
@@ -635,4 +635,34 @@ mod test {
         );
         Ok(())
     }
+
+    #[test]
+    fn instantiate_extension_delta() -> Result<(), Box<dyn std::error::Error>> {
+        use crate::extension::prelude::{BOOL_T, PRELUDE_REGISTRY};
+
+        let mut e = Extension::new(EXT_ID);
+        let ext_id = e.name().clone();
+
+        let es = ExtensionSet::singleton(&ext_id);
+        let params: Vec<TypeParam> = vec![TypeParam::Extensions];
+        let args = [TypeArg::Extensions { es: es.clone() }];
+        let mut db_set = ExtensionSet::new();
```

Comment:
nit: can you use `let db_set = ExtensionSet::type_var(0)`? Or, include a non-variable too (so `let mut ... = ExtensionSet::singleton` followed by `insert_type_var`) and check we take the union?